ReactAndroid/src/main/java/com/facebook/react/views/switchview/ReactSwitchManager.java

```diff
@@ -175,6 +175,12 @@ public void setTrackTintColor(ReactSwitch view, @Nullable Integer color) {
     view.setTrackColor(color);
   }
 
+  @ReactProp(name = "androidMinWidth")
+  public void setSwitchMinWidth(ReactSwitch view, Integer pixels) {
+    Integer width = pixels == null ? view.getSwitchMinWidth() : pixels;
+    view.setSwitchMinWidth(width);
```

Comment:
I just have a comment (or question). React-Native uses dip (density independent pixels) for the width and size of the components., as stated in the docs: [https://reactnative.dev/docs/height-and-width](https://reactnative.dev/docs/height-and-width).

Shouldn't this method expect a dip value instead of pixels ? The conversion from dip to pixels can be handled using this method:

```java
  private float dpToPixels(float dp) {
    return TypedValue.applyDimension(
      TypedValue.COMPLEX_UNIT_DIP,
      dp,
      view.getContext().getResources().getDisplayMetrics()
    );
  }
```

Reply:
@HectorRicardo 

I added https://github.com/facebook/react-native/pull/29059/commits/7c6901e72b511565bd2ce2c1b6d91de2ef2f1956 to use 

I quote the [dpi definition](https://en.wikipedia.org/wiki/Device-independent_pixel)

>A device-independent pixel (also: density-independent pixel, dip, dp) is a unit of length.

>For example, on the Android operating system a device-independent pixel is equivalent to one physical pixel on a 160 dpi screen,[1] while the Windows Presentation Foundation specifies one device-independent pixel as equivalent to 1/96th of an inch.[2]

>As **dp** is a physical unit it has an absolute value which can be measured in traditional units, e.g. for Android devices 1 dp equals 1/160 of inch or 0.15875 mm.

Increasing `density` of pixels does not change size

![image](https://user-images.githubusercontent.com/24992535/83855156-fa152e80-a717-11ea-8388-025a67b55813.png)

`setSwitchMinWidth` specifies the `MINUMUM WIDTH`. The actual width will be specified with the [`style` prop](https://reactnative.dev/docs/view-style-props). The `style` prop will not be applied if the `minWidth` is not respected.


| **Density 300 - switch width 50 minWidth 50** | **Density 300 - switch width 50 minWidth 100** |
|:-------------------------:|:-------------------------:|
| <img src="https://user-images.githubusercontent.com/24992535/83858014-1d41dd00-a71c-11ea-9dd1-a0c7e1b290b9.png" width="250" height="" /> | <img src="https://user-images.githubusercontent.com/24992535/83858062-3054ad00-a71c-11ea-80ee-c3412cbf07e4.png" width="250" height="" /> |

Persisting Issues with the actual width of this component could be related to https://github.com/facebook/react-native/issues/28235
In that case I would write a separate pull request to fix that dpi issue and fix the following PixelUtil or the View Style prop.

https://github.com/facebook/react-native/blob/851d01b0aa45285bd6a52c89ab9c1b64a637258e/ReactAndroid/src/main/java/com/facebook/react/uimanager/PixelUtil.java#L13-L20

| **Density 200 - switch is 50px width** | **Density 200 - switch is 200px width** |
|:-------------------------:|:-------------------------:|
| <img src="https://user-images.githubusercontent.com/24992535/83855752-d30b2c80-a718-11ea-8de1-312ddeb9d2de.png" width="250" height="" /> | <img src="https://user-images.githubusercontent.com/24992535/83856006-32693c80-a719-11ea-98f9-e3f957dd1b17.png" width="250" height="" /> |

| **Density 300 - switch is 50px width** | **Density 300 - switch is 200px width** |
|:-------------------------:|:-------------------------:|
| <img src="https://user-images.githubusercontent.com/24992535/83856105-5f1d5400-a719-11ea-8b1d-4fa75680bb85.png" width="250" height="" /> | <img src="https://user-images.githubusercontent.com/24992535/83856160-6f353380-a719-11ea-9ad3-cf46d849dab1.png" width="250" height="" /> |

Reply:
Thanks for the explanation Fabrizio! It makes total sense what you're saying, that the style's width will not be applied if min width is not respected.